Remove hardcoded favicon.

samkelleher opened the pull request:
Fixes #921.
Fixes #216.

Changes proposed in this pull request:

- Remove the hardcoded favicon `graphcool-playground.netlify.com` path.
This hardcoded path never actually did anything as it was replaced by a further `shortcut icon` link after, so is being removed for hygiene.

Before:
```
<link rel="shortcut icon" href="https://graphcool-playground.netlify.com/favicon.png">
<title>GraphQL Playground</title>
<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/graphql-playground-react@1.7.8/build/static/css/index.css" />
<link rel="shortcut icon" href="//cdn.jsdelivr.net/npm/graphql-playground-react@1.7.8/build/favicon.png" />
```
After:
```
<title>GraphQL Playground</title>
<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/graphql-playground-react@1.7.8/build/static/css/index.css" />
<link rel="shortcut icon" href="//cdn.jsdelivr.net/npm/graphql-playground-react@1.7.8/build/favicon.png" />
```

Huvik commented:
Hey @samkelleher thanks for PR, but this way we are going to remove default favicon all together. So it could use different cdn #845 or something like this. So options are:
* Use cdn url
* Provide custom url by props

samkelleher commented:
@Huvik , A pleasure to submit as is only a small change.  Can you check as there is already a default icon being set using the correct configurable CDN address which uses `cdn.jsdelivr.net` as default, so there is always an icon. See the script in the file.

The removed line is hard coded and is never actually called (unless the icon on the CDN path fails to load).

Huvik commented:
@samkelleher Okey, I am blind 😃, so I think custom URL overriding this is the best idea.

samkelleher commented:
Is there any action for me to take on this then @Huvik  - the custom CDN url is already present. This PR removes the redundant line that isn't used.

Huvik commented:
Sorry maybe my response wasn't clear, but it would be really cool to provide custom favicon to html page. If we provide custom favicon for html page we can override cdn favicon and have it options for middlewares too. What do you think about it?

samkelleher commented:
@Huvik  okay I've added a config option to provide a custom favicon URL. The behaviour is as follows.
_No config provided_, will use the favicon hosted on the CDN (the CDN is configurable as before), but the hard-coded unconfigurable favicon is still removed.
_When a config is provided_, by middleware, the user can provide a path to a custom favicon of their choice.

I've also added the type key for title, which wasn't referenced.

Express Middleware Example:
```
import express from 'express';
import expressGraphQLPlayground from 'graphql-playground-middleware-express';
import config from './config';

const server = express();

server.use(
    '/playground',
    expressGraphQLPlayground({
        endpoint: '/graphql',
        title: `GraphQL Playground - ${config.environment.name}`,
        cdnUrl: '/playground-assets',
        faviconUrl: '/favicon.ico',
    }),
);
```

Huvik commented:
Nice! That looks good to go 🚀I will test it tomorrow and if everything is okay it will be in next release!

Huvik commented:
I just recall there is issue https://github.com/prisma/graphql-playground/issues/216 maybe we could implement this too or do we want just add favicon like props? Maybe we could add another props, what do you think?

samkelleher commented:
Have pushed commit that will fix #216

To use a custom favicon, provide path:

```
expressGraphQLPlayground({ faviconUrl: '/favicon.ico' });
```

To use the default favicon provided by default CDN.

```
expressGraphQLPlayground({ });
```

To not add a shortcut icon whatsoever (thus falling back to the browser behaviour for finding the favicon, (ie the favicon.ico in the root)
```
expressGraphQLPlayground({ faviconUrl: null }); // Can also set faviconUrl to false
```

Huvik commented:
Great job! 🚀

Huvik commented:
Thanks for PR! It will be in next release. 🚀
